src/DI/PhpReflection.php

```diff
@@ -81,6 +87,9 @@ public static function expandClassName($name, \ReflectionClass $rc)
 		if (empty($name)) {
 			throw new Nette\InvalidArgumentException('Class name must not be empty.');
 
+		} elseif (in_array($name, PHP_VERSION_ID >= 70000 ? ['string', 'int', 'float', 'bool', 'array', 'callable'] : ['array', 'callable'], TRUE)) {
```

Comment:
I think that version checking is not required here, so `in_array($name, ['string', 'int', 'float', 'bool', 'array', 'callable'], TRUE)` is enough.